Content.Server/GameObjects/Components/Movement/ClimbableComponent.cs

```diff
@@ -60,68 +61,101 @@ public override void ExposeData(ObjectSerializer serializer)
+                reason = Loc.GetString("You are incapable of climbing!");
+                return false;
+            }
 
-                var bodyManager = eventArgs.User.GetComponent<BodyManagerComponent>();
+            var bodyManager = user.GetComponent<BodyManagerComponent>();
 
-                if (bodyManager.GetBodyPartsOfType(Shared.GameObjects.Components.Body.BodyPartType.Leg).Count == 0 ||
-                    bodyManager.GetBodyPartsOfType(Shared.GameObjects.Components.Body.BodyPartType.Foot).Count == 0)
-                {
-                    _notifyManager.PopupMessage(eventArgs.User, eventArgs.User, Loc.GetString("You are unable to climb!"));
+            if (bodyManager.GetBodyPartsOfType(Shared.GameObjects.Components.Body.BodyPartType.Leg).Count == 0 ||
+                bodyManager.GetBodyPartsOfType(Shared.GameObjects.Components.Body.BodyPartType.Foot).Count == 0)
+            {
+                reason = Loc.GetString("You are unable to climb!");
+                return false;
+            }
 
-                    return false;
-                }
+            var userPosition = user.Transform.MapPosition;
+            var climbablePosition = target.Transform.MapPosition;
+            var interaction = EntitySystem.Get<SharedInteractionSystem>();
+            bool Ignored(IEntity entity) => (entity == target || entity == user);
 
-                var userPosition = eventArgs.User.Transform.MapPosition;
-                var climbablePosition = eventArgs.Target.Transform.MapPosition;
-                var interaction = EntitySystem.Get<SharedInteractionSystem>();
-                bool Ignored(IEntity entity) => (entity == eventArgs.Target || entity == eventArgs.User);
+            if (!interaction.InRangeUnobstructed(userPosition, climbablePosition, _range, predicate: Ignored))
+            {
+                reason = Loc.GetString("You can't reach there!");
+                return false;
+            }
 
-                if (!interaction.InRangeUnobstructed(userPosition, climbablePosition, _range, predicate: Ignored))
-                {
-                    _notifyManager.PopupMessage(eventArgs.User, eventArgs.User, Loc.GetString("You can't reach there!"));
+            reason = string.Empty;
+            return true;
+        }
 
-                    return false;
-                }
-            }
-            else // user is dragging some other entity onto a climbable
+        /// <summary>
+        /// Checks if the user can vault the dragged entity onto the the target
+        /// </summary>
+        /// <param name="user">The user that wants to vault the entity</param>
+        /// <param name="dragged">The entity that is being vaulted</param>
+        /// <param name="target">The object that is being vaulted onto</param>
+        /// <param name="reason">The reason why it cant be dropped</param>
+        /// <returns></returns>
+        private bool CanVault(IEntity user, IEntity dragged, IEntity target, out string reason)
```

Comment:
This has duplicate checks with the other CanVault method, is it possible to combine these a bit more?

Reply:
Mhh you I make a pre function that will take dragged as null and then check the first two things, but I don't think that's better

Reply:
Yeah I'm not sure what it should be; you could try changing the internals so they both call a 3rd function with the common functionality, but you could just out of scope it for the pr and leave as is I guess.